pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -135,7 +136,17 @@ func (s *Server) initDNSCerts(hostname, customHost, namespace string) error {
 
 	var certChain, keyPEM []byte
 	var err error
+
 	if features.PilotCertProvider.Get() == KubernetesCAProvider {
+		var k8sVersion string
+		pilotUseK8sCA, k8sVersion = kubeDefaultSignerSupported(s.kubeClient)
+		if !pilotUseK8sCA {
+			log.Errorf("Unable to sign DNS certificates using Chiron in current deployment. "+
+				"Kubernetes legacy signer is not supported from Kubernetes version 1.%s.0 onwards. "+
+				"Falling back to using istiod to sign DNS certificate", k8sVersion)
```

Comment:
I don't think fallback is really safe? We should probably fail closed on security related matters.